src/client/inputhandler.h

```diff
@@ -192,6 +211,14 @@ class RealInputHandler : public InputHandler
 	{
 		return m_receiver->WasKeyDown(keyCode);
 	}
+	virtual void listenForKey(const KeyPress &keyCode)
```

Comment:
Absolutely no, you inverted the inheritance concept. You are in RealInputHandler, inheriting from InputHandler you version will be the correct version. virtual should be dropped there.